modules/watson-machine-learning/variables.tf

```diff
@@ -8,6 +8,38 @@ variable "cos_instance_name" {
   type        = string
 }
 
+variable "cos_kms_crn" {
+  description = "Key Protect service instance CRN used to encrypt the COS buckets used by the watsonx projects."
+  type        = string
+  default     = null
+
+  validation {
+    condition = anytrue([
+      can(regex("^crn:(.*:){3}kms:(.*:){2}[0-9a-fA-F]{8}(?:-[0-9a-fA-F]{4}){3}-[0-9a-fA-F]{12}::$", var.cos_kms_crn)),
```

Comment:
This restricts it to Key Protect, but COS supports HPCS encryption too

Reply:
I took this directly from the WatsonX Saas DA [here](https://github.com/terraform-ibm-modules/terraform-ibm-watsonx-saas-da/blob/main/storage_delegation/variables.tf#L21) - I can change it but they'll need to update as well as the validation would fail in the storage-delegation module regardless of my changes.

I was also under the impression that HPCS is going away soon - @in-1911 any thoughts on this one?

Reply:
@ocofaigh - Watson storage delegation only supports KeyProtect - so that validation just reflects what they have now. 
If Watson services start to support HPCS, we can adjust that.